blocks/algorithms/__init__.py

```diff
@@ -222,6 +224,29 @@ def compute_step(self, param, grad_wr_param):
         """
         raise NotImplementedError
 
+    def compute_steps(self, gradients):
+        """Build a Theano expression for steps for all parameters.
+
+        Override this method if you want to process the gradient
+        with respect to all parameters as a whole, not parameter-wise.
+
+        Parameters
+        ----------
+        gradients : dict of (:class:`~tensor.TensorSharedVariable`
```

Comment:
But in `initialize` (lines 189 and 190) the updates are added in the deterministic order, isn't that enough?